elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -413,23 +413,30 @@ export class PeerPool extends EventEmitter {
 			disconnectedFixedPeers.length -
 			outboundCount;
 
+		const bestPeersbyHeight = getBestPeersbyHeight([
+			...disconnectedNewPeers,
+			...disconnectedTriedPeers,
+			...disconnectedFixedPeers,
+		]);
+
 		// This function can be customized so we should pass as much info as possible.
 		const peersToConnect = this._peerSelectForConnection({
-			newPeers: disconnectedNewPeers,
-			triedPeers: disconnectedTriedPeers,
+			newPeers: comparePeerListsByHeight(
+				disconnectedNewPeers,
+				bestPeersbyHeight,
+			),
+			triedPeers: comparePeerListsByHeight(
+				disconnectedTriedPeers,
+				bestPeersbyHeight,
+			),
 			nodeInfo: this._nodeInfo,
 			peerLimit,
 		});
 
-		[...peersToConnect, ...disconnectedFixedPeers].forEach(
-			(peerInfo: P2PPeerInfo) => {
-				const existingPeer = this.getPeer(peerInfo.peerId);
-
-				return existingPeer
-					? existingPeer
-					: this.addOutboundPeer(peerInfo.peerId, peerInfo);
-			},
-		);
+		comparePeerListsByHeight(
```

Comment:
Only use the function here `getBestPeersbyHeight([...peersToConnect, ...disconnectedFixedPeers])`. I think its enough